lib/experimental/interactivity-api/directive-processing.php

```diff
@@ -19,120 +19,43 @@
  * @return array The parsed block.
  */
 function gutenberg_interactivity_process_directives( $parsed_block, $source_block, $parent_block ) {
-	static $is_inside_root_block              = false;
-	static $process_directives_in_root_blocks = null;
-
-	if ( ! isset( $process_directives_in_root_blocks ) ) {
-		/**
-			 * Process directives in each root block.
-			 *
-			 * @param string $block_content The block content.
-			 * @param array  $block         The full block.
-			 *
-			 * @return string Filtered block content.
-			 */
-		$process_directives_in_root_blocks = static function ( $block_content, $block ) use ( &$is_inside_root_block ) {
-
-			if ( WP_Directive_Processor::is_root_block( $block ) ) {
-
-				$directives = array(
-					'data-wp-bind'    => 'gutenberg_interactivity_process_wp_bind',
-					'data-wp-context' => 'gutenberg_interactivity_process_wp_context',
-					'data-wp-class'   => 'gutenberg_interactivity_process_wp_class',
-					'data-wp-style'   => 'gutenberg_interactivity_process_wp_style',
-					'data-wp-text'    => 'gutenberg_interactivity_process_wp_text',
-				);
-
-				$tags                 = new WP_Directive_Processor( $block_content );
-				$tags                 = gutenberg_interactivity_process_rendered_html( $tags, 'data-wp-', $directives );
-				$is_inside_root_block = false;
-				return $tags->get_updated_html();
-
-			}
-
-			return $block_content;
-		};
-		add_filter( 'render_block', $process_directives_in_root_blocks, 10, 2 );
-	}
-
-	if ( ! isset( $parent_block ) && ! $is_inside_root_block ) {
+	if ( ! isset( $parent_block ) && ! isset( WP_Directive_Processor::$root_block ) ) {
```

Comment:
I'd rather use `WP_Directive_Processor::has_root_block()` to avoid exposing the property publicly.